websites/ubuntu.lt/templates/ubuntult.service

```diff
@@ -8,9 +8,9 @@ Type=notify
 User={{ slug }}
 Group=www-data
 RuntimeDirectory={{ slug }}
-WorkingDirectory={{ home }}
+WorkingDirectory={{ path }}
 EnvironmentFile={{ home }}/.env
-ExecStart={{ home }}/venv/bin/gunicorn -w 4 -b 127.0.0.1:8120 project.wsgi.prod
+ExecStart={{ home }}/venv/bin/gunicorn -w 4 project.wsgi.prod
```

Comment:
`127.0.0.1:8120` is expected in https://github.com/akllt/infrastructure/blob/master/websites/ubuntu.lt/templates/apache.conf#L53-L54

Reply:
For systemd deployments the listening socket address is specified in the `.socket` file, not on the command line here.

Reply:
If you're curious how that works, see https://www.freedesktop.org/software/systemd/man/systemd.socket.html.  Basically

- you tell systemd what ports to open
- when something connects to that port, systemd starts the associated .service unit
- when starting a service unit, systemd sets a couple of environment variables (LISTEN_FDS and LISTEN_PID)
- gunicorn [reads those environment variables](https://github.com/benoitc/gunicorn/blob/master/gunicorn/systemd.py#L12-L46) and knows which file descriptors, inherited from the parent process (i.e. systemd), contain the listening socket(s).

Reply:
Thanks, today I learned a new thing.